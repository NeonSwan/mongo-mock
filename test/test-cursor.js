var cursor = require("../lib/mongo-mock/cursor"); // Internal API!

var c = cursor([1, 2, 3]);
console.log("cursor([1, 2, 3]).each:");
c.each(function(error, object) {
  console.log("  " + object);
});
console.log("");

var c = cursor([1, 2, 3]);
console.log("cursor([1, 2, 3]).nextObject:");
c.nextObject(function(error, object) {
  console.log("  " + object);
  c.nextObject(function(error, object) {
    console.log("  " + object);
    c.nextObject(function(error, object) {
      console.log("  " + object);
        c.nextObject(function(error, object) {
          console.log("  " + object);
        });
    });
  });
});
console.log("");

var c = cursor([1, 2, 3]);
console.log("cursor([1, 2, 3]).toArray:");
c.toArray(function(error, objects) {
  console.log("  " + objects);
});
console.log("");
