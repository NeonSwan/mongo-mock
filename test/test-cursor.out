cursor([1, 2, 3]).each:
  1
  2
  3
  null

cursor([1, 2, 3]).nextObject:
  1
  2
  3
  null

cursor([1, 2, 3]).toArray:
  1,2,3

